src/targets/gpu/include/migraphx/gpu/compile_scatternd.hpp

```diff
@@ -0,0 +1,20 @@
+#ifndef MIGRAPHX_GUARD_GPU_COMPILE_SCATTERND_HPP
+#define MIGRAPHX_GUARD_GPU_COMPILE_SCATTERND_HPP
+
+#include <migraphx/config.hpp>
+#include <migraphx/operation.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct context;
+operation
+compile_scatternd(context& ctx, const std::vector<shape>& io_shapes, const std::string& reduction);
```

Comment:
Just set a default parameter for `reduction` to `none` instead of having two overloads.